noir-projects/aztec-nr/aztec/src/oracle/message_discovery.nr

```diff
@@ -62,14 +60,15 @@ pub struct LogWithTxData {
     pub first_nullifier_in_tx: Field,
 }
 
-/// Fetches a log from the node that has the corresponding `tag`. The log can be either a public or a private log, and
-/// the tag is the first field in the log's content. Returns `Option::none` if no such log exists. Throws if more than
-/// one log with that tag exists.
-/// Public logs have an extra field included at the beginning with the address of the contract that emitted them.
+/// Fetches a public log from the node that has the corresponding `tag`. The tag is the first field in the log's content.
+/// Returns `Option::none` if no such log exists. Throws if more than one log with that tag exists.
 // TODO(#11627): handle multiple logs with the same tag.
 // TODO(#10273): improve contract siloing of logs, don't introduce an extra field.
-pub unconstrained fn get_log_by_tag(tag: Field) -> Option<LogWithTxData> {
-    get_log_by_tag_oracle(tag)
+pub unconstrained fn get_public_log_by_tag_for_contract(
```

Comment:
maybe '_from_contract`?

Reply:
Addressed in https://github.com/AztecProtocol/aztec-packages/pull/14459/commits/641338389cd631d473efddb28ac3c16f7434c1f6